### PR TITLE
[BEAM-9442] Properly handle nullable fields in Select

### PR DESCRIPTION
**Pull request by reuvenlax**
Fix the selected schema to properly represent nullable nested fields.

Fix the schema code to check for null values and not throw NullPointerExceptions.

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/SelectByteBuddyHelpers.java [hunk 1]
```diff
@@ -256,6 +260,10 @@ public StackManipulation append(StackManipulation valueToWrite) {
       return store(currentArrayField++, valueToWrite);
     }
 
+    public int reserveSlot() {
```
Comment (iemejia) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/SelectByteBuddyHelpers.java]: nit: (slightly related to this PR) can we make `SelectByteBuddyHelpers` package private please

Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/SelectByteBuddyHelpers.java]: done

**Comment by reuvenlax** [comment]
Run Spotless PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by alexvanboxel** [comment]
The error in the test is my fault, approved a fix without seeing the tests were ran. Just need to be rebased.

**Comment by reuvenlax** [comment]
@alexvanboxel are you talking about the RabbitMQ failure?

**Comment by alexvanboxel** [comment]
> @alexvanboxel are you talking about the RabbitMQ failure?

yes (rookie, mistake of me)

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
friendly ping: @alexvanboxel @iemejia

**Comment by alexvanboxel** [comment]
@iemejia wasn't available this week. His remark are fixed. So LGTM

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Pull request merged**